test-suite/lazyobject.cpp

```diff
@@ -58,13 +58,11 @@ void LazyObjectTest::testDiscardingNotifications() {
 void LazyObjectTest::testForwardingNotifications() {
 
     BOOST_TEST_MESSAGE(
-        "Testing that lazy objects forward all notifications when told...");
+        "Testing that lazy objects forward all notifications by default...");
```

Comment:
I would also call `forwardFirstNotificationOnly` and check the corresponding behavior (either in another test, or making this one longer.)